src/helpers/fallbackauthdialog.js

```diff
@@ -65,6 +94,69 @@ function handleResponse(response) {
     localStorage.setItem("loggedIn", "yes");
   } else {
     console.log("token type is:" + response.tokenType);
-    Office.context.ui.messageParent(JSON.stringify({ status: "success", result: response.accessToken }));
+    Office.context.ui.messageParent(
+      JSON.stringify({ status: "success", result: response.accessToken, accountId: response.account.homeAccountId })
+    );
+  }
+}
+
+export async function dialogFallback(callback) {
+  // Attempt to acquire token silently if user is already signed in.
+  if (homeAccountId !== null) {
+    const result = await publicClientApp.acquireTokenSilent(loginRequest);
+    if (result !== null && result.accessToken !== null) {
+      const response = await callGetUserData(result.accessToken);
+      callbackFunction(response);
+    }
+  } else {
+    callbackFunction = callback;
+
+    // We fall back to Dialog API for any error.
+    const url = "/fallbackauthdialog.html";
+    showLoginPopup(url);
   }
 }
+
+// This handler responds to the success or failure message that the pop-up dialog receives from the identity provider
+// and access token provider.
+async function processMessage(arg) {
+  console.log("Message received in processMessage: " + JSON.stringify(arg));
+  let messageFromDialog = JSON.parse(arg.message);
+
+  if (messageFromDialog.status === "success") {
+    // We now have a valid access token.
+    loginDialog.close();
+
+    // Configure MSAL to use the signed-in account as the active account for future requests.
+    const homeAccount = publicClientApp.getAccountByHomeId(messageFromDialog.accountId);
+    if (homeAccount) {
+      homeAccountId = messageFromDialog.accountId; // Track the account id for future silent token requests.
+      publicClientApp.setActiveAccount(homeAccount);
+    }
+
+    const response = await callGetUserData(messageFromDialog.result);
+    callbackFunction(response);
+  } else if (messageFromDialog.error === undefined && messageFromDialog.result.errorCode === undefined) {
+    // Need to pick the user to use to auth
```

Comment:
I added this to the typescript template (this is a port of that change) when I encountered a case where the status was not "success" and there was no error reported.  It appeared that this would happen when an account choice list for auth needed to be presented to the user in the dialog, but the dialog would get closed (because it fell into the else block) before that could happen.  It's possible that something else may have changed since then to prevent this case from being needed, but I haven't tried that out.

Reply:
OK.